contracts/dependency-registry/DependencyRegistryV1.sol

```diff
@@ -0,0 +1,1123 @@
+                ).projectScriptByIndex(_projectId, i);
+                script = string.concat(script, scriptChunk);
+            }
+
+            return abi.encodePacked(script);
+        } catch {
+            revert("Unable to retrieve project script info");
+        }
+    }
+
+    function getTokenHtmlScriptRequests(
+        address _contractAddress,
+        uint256 _tokenId
+    )
+        internal
+        view
+        returns (WrappedScriptRequest[] memory, uint256 bufferSize)
+    {
+        _onlySupportedCoreContract(_contractAddress);
+
+        uint256 projectId = _tokenId / ONE_MILLION;
+        // This will revert for older contracts that do not have an override set.
+        bytes32 dependencyType = stringToBytes32(
+            this.getDependencyTypeForProject(_contractAddress, projectId)
+        );
+
+        bytes32 tokenHash;
+        try
+            IDependencyRegistryCompatibleV0(_contractAddress).tokenIdToHash(
+                _tokenId
+            )
+        returns (bytes32 _tokenHash) {
+            tokenHash = _tokenHash;
+        } catch {
+            // Noop try again for older contracts.
+        }
+
+        if (tokenHash == bytes32(0)) {
+            try
+                IGenArt721CoreTokenHashProviderV1(_contractAddress)
+                    .tokenIdToHash(_tokenId)
+            returns (bytes32 tokenHash) {
+                tokenHash = tokenHash;
+            } catch {
+                // Noop try again for older contracts.
+            }
+        }
+
+        if (tokenHash == bytes32(0)) {
+            try
+                IGenArt721CoreTokenHashProviderV0(_contractAddress)
+                    .showTokenHashes(_tokenId)
+            returns (bytes32[] memory tokenHashes) {
+                tokenHash = tokenHashes[0];
+            } catch {
+                revert("Unable to retrieve token hash.");
+            }
+        }
+
+        WrappedScriptRequest[] memory requests = new WrappedScriptRequest[](5);
+
+        requests[0]
+            .scriptContent = 'let css="html{height:100%}body{min-height:100%;margin:0;padding:0}canvas{padding:0;margin:auto;display:block;position:absolute;top:0;bottom:0;left:0;right:0}",head=document.head,style=document.createElement("style");head.appendChild(style),style.type="text/css",style.appendChild(document.createTextNode(css));';
+
+        requests[1].scriptContent = abi.encodePacked(
+            'let tokenData = {"tokenId":"',
+            Strings.toString(_tokenId),
+            '"',
+            ',"hash":"',
+            Strings.toHexString(uint256(tokenHash)),
+            '"}'
+        );
+
+        bytes memory dependencyScript = this.getDependencyScript(
+            dependencyType
+        );
+        requests[2].scriptContent = dependencyScript;
+        requests[2].wrapType = 2; // <script type="text/javascript+gzip" src="data:text/javascript;base64,[script]"></script>
+
+        bytes memory gunzipScript = IContractScript(ethFsAddress).getScript(
```

Comment:
@jakerockland @ryley-o Are we good with gzipping our dependency scripts? This should allow the project scripts to be longer before hitting any provider enforced gas limit so seems like a good idea to me. Also, how do we feel about depending on ethfs for pulling this script?